File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        String first = s.substring(0,1);
        int firstCnt = 0;
        int anotherCnt =0;
        
        String[] chars = s.split("");
        for(int i=0; i<chars.length; i++){
            String ch = chars[i];
            if(first.equals(ch)){
                firstCnt++;
            }else{
                anotherCnt++;
            }
            if(firstCnt == anotherCnt){
                answer++;
                firstCnt = 0;
                anotherCnt = 0;
                if(i<chars.length-1) first = chars[i+1];
            }
        }
        if(firstCnt !=0 || anotherCnt!=0) answer++;
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    let first = "init";
    let firstCnt = 0;
    let anotherCnt = 0;
    
    let chars = s.split("");
    
    for(let ch of chars){
        if(first === "init"){
            first = ch;
            firstCnt++;
        }else if(first === ch){
            firstCnt++;
        }else{
            anotherCnt++;
        }
        
        if(firstCnt === anotherCnt){
            answer++;
            firstCnt = 0;
            anotherCnt = 0;
            first = "init";
        }
    }
    if(firstCnt !=0 || anotherCnt !=0) answer++;
    return answer;
}